src/bm_sim/switch.cpp

```diff
@@ -72,6 +73,8 @@ SwitchWContexts::receive(port_t port_num, const char *buffer, int len) {
 
 void
 SwitchWContexts::start_and_return() {
+  // Starts any registered periodically-executing externs
+  PeriodicTaskList::get_instance().start();
```

Comment:
Ir would probably beet to put this with the `start` call underneath, unless you have a good reason to start execution before a P4 program is loaded?